internal/msalbase/token_response.go

```diff
@@ -30,6 +30,8 @@ type tokenResponseJSONPayload struct {
 type ClientInfoJSONPayload struct {
 	UID  string `json:"uid"`
 	Utid string `json:"utid"`
+
+	AdditionalFields map[string]interface{}
```

Comment:
Can we get a test?

Reply:
@keegan-caruso 

I was thinking about eventually removing the existing tests serialization tests.  I left the existing ones in to validate that the encoder/decoder was doing what it should outside its internal tests when I switched it out.

What I wanted to have before that was an etoe that proved that these worked.  The main bug here looks like that this was never used in any tests and of course the first time it was used, it croaked because the new library requires all structs to have AdditionalFields.  

Do you think we need these serialization/deserialization once we are actually testing the flows?  It provides extra safety, but I've always felt these types of tests were excessive and are usually a sign that the etoe and unit tests are weak.

Reply:
It sounds wrong to have code here that is not exercised at all in UT. Agreed that e2e should catch it, but I'd prefer to have quicker feedback loops. e2e usually doesn't get run often enough.

That might be part of a larger discussion scope than these changes though.

Reply:
I agree that there should be a unit test of whatever is using this message, which is not happening here.  But I want to separate the capability unit test that doesn't exist (something that uses this message) from a test for just encoding/decoding.  

Specifically,  do we need encoding/decoding tests vs just usage tests and etoe coverage?  We can talk this talk offline.